File: sdk-server/modules/sys/service/sys_sms.go
package service

import (
	"dilu/common/config"
	"dilu/common/third/sms"
	"dilu/modules/sys/models"
	"fmt"
	"sync"
	"time"

	"github.com/baowk/dilu-core/common/consts"
	"github.com/baowk/dilu-core/core"
	"github.com/baowk/dilu-core/core/base"
)

type SysSms struct {
	*base.BaseService
}

var SerSms = SysSms{
	base.NewService(consts.DB_DEF),
}

var onceSms sync.Once
var smsSend sms.SmsSend

func (e *SysSms) getSms() sms.SmsSend {
	onceSms.Do(func() {
		smsSend = sms.Setup(&config.Ext.Sms)
	})
	return smsSend
}

func (e *SysSms) Send(phone string, tempId string) error {
	var err error
	code := sms.GenerateSmsCode(6)
	data := models.SysSms{
		Phone:     phone,
		Code:      code,
		Type:      "Code",
		Status:    0,
		UseStatus: 0,
	}
	data.CreatedAt = time.Now().Unix()
	data.UpdatedAt = data.CreatedAt
	err = core.DB().Create(&data).Error
	if err != nil {
		return err
	}
	return e.getSms().Send(phone, code, tempId)
}

// 验证
func (e *SysSms) Verify(phone, code string) bool {
	if core.Cfg.Server.Mode != "prod" && code == "666666" {
		return true
	}
	var err error
	var data models.SysSms
	err = core.DB().Model(&data).Where("phone = ? ", phone).Order("id desc").Find(&data).Error
	if err != nil {
		return false
	}
	if data.Id == 0 {
		fmt.Println("没有找到验证码", phone, code)
		return false
	}
	if data.UseStatus == 1 {
		fmt.Println("已使用")
		return false
	}
	if data.CreatedAt+int64(time.Minute.Minutes()*10) < time.Now().Unix() {
		fmt.Println("1111")
		if code == data.Code {
			fmt.Println("111333")
			data.UseStatus = 1
			data.UpdatedAt = time.Now().Unix()
			core.DB().Save(&data)
			return true
		}
	}

	return false
}

File: sdk-admin/src/api/sys/sys-sms.ts
import { http } from "@/utils/http";
import { Result, PageResult } from "@/api/common";
//import { SysSms } from "@/api/sys/sys-sms.d";

//Api 

/** 获取短信管理列表 */
export const getSysSmsPage = (data?: object) => {
  return http.request<Result<PageResult<SysSms>>>(
    "post",
    "/api/sys/sys-sms/page",
    {
      data
    }
  );
};

/** 获取短信 */
export const getSysSms = (data?: object) => {
  return http.request<Result<SysSms>>("post", "/api/sys/sys-sms/get", {
    data
  });
};

/** 创建短信 */
export const createSysSms = (data?: object) => {
  return http.request<Result<SysSms>>("post", "/api/sys/sys-sms/create", {
    data
  });
};

/** 更新短信 */
export const updateSysSms = (data?: object) => {
  return http.request<Result<SysSms>>("post", "/api/sys/sys-sms/update",{
    data
  });
};

/** 删除短信 */
export const delSysSms = (data?: object) => {
  return http.request<Result<SysSms>>("post", "/api/sys/sys-sms/del",{
    data
  });
};

//Model

/** 短信 */
interface SysSms {
  
  /** 主键编码 */
  id: number;
  /** 手机号 */
  phone: string;
  /** 验证码 */
  code: string;
  /** 类型 */
  type: string;
  /** 状态 */
  status: number;
  /** 使用状态 */
  useStatus: number;
  /** 创建时间 */
  createdAt: number;
  /** 最后更新时间 */
  updatedAt: number;
}


interface SysSmsFormItemProps {
  
  /** 主键编码 */
  id: number;
  /** 手机号 */
  phone: string;
  /** 验证码 */
  code: string;
  /** 类型 */
  type: string;
  /** 状态 */
  status: number;
  /** 使用状态 */
  useStatus: number;
}
interface SysSmsFormProps {
  formInline: SysSmsFormItemProps;
}

export type { SysSmsFormItemProps, SysSmsFormProps, SysSms };